components/provisioner/internal/operations/executor.go

```diff
@@ -115,6 +115,10 @@ func (e *Executor) process(operation model.Operation, cluster model.Cluster, log
 
 		result, err := step.Run(cluster, operation, log)
 		if err != nil {
+			if err.Error() == "cluster kubeconfig is nil" {
```

Comment:
I don't think this is the recommended way to differentiate errors since Go 1.13
Maybe this code should use `errors.Is`/`errors.As`?
https://pkg.go.dev/errors